src/main/java/core/mate/academy/service/impl/ExcavatorProducerImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.mate.academy.service.impl;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducerImpl implements MachineProducer<Excavator> {
+    private final List<Excavator> excavators;
```

Comment:
The field 'excavators' should have an access modifier. Preferably private to encapsulate the data.